central/vulnmgmt/vulnerabilityrequest/manager/requestmgr/manager_impl.go

```diff
@@ -577,21 +579,36 @@ func (m *managerImpl) getFixableDeferrals() ([]*storage.VulnerabilityRequest, er
 	}
 	var fixableReqs []*storage.VulnerabilityRequest
 	for _, res := range results {
-		for _, cve := range res.GetCves().GetCves() {
+		for _, CVE := range res.GetCves().GetCves() {
 			qb := search.NewQueryBuilder().AddBools(search.Fixable, true)
 			// TODO: Determine if it's worth checking cvePkg.ContainsComponentBasedCVE(cve.GetTypes()) before doing this. It would involve going to the data store to read CVE data
 			// This is only necessary if somehow there ended up being a deferral on a cluster CVE. Or if it goes from an image cve to node cve
 			// TODO: Test what happens if it's a cluster cve
 			ctx := allAccessCtx
-			qb.AddExactMatches(search.CVE, cve)
+			qb.AddExactMatches(search.CVE, CVE)
 
 			fixableQuery, err := utils.GetAffectedImagesQuery(res, qb.ProtoQuery())
 			if err != nil {
 				return nil, err
 			}
-			count, err := m.imageCVEs.Count(ctx, fixableQuery)
-			if err != nil {
-				return nil, errors.Wrapf(err, "could not fetch cve component edge for cve %q for request %q", cve, res.GetId())
+			var count int
+			if features.FlattenCVEData.Enabled() {
+				resultCVEs, err := m.imageCVEsV2.Search(ctx, fixableQuery)
+				if err != nil {
+					return nil, errors.Wrapf(err, "could not search image cves for request %q", res.GetId())
+				}
+				cveSet := set.NewStringSet()
+				for _, id := range search.ResultsToIDs(resultCVEs) {
+					cve, _ := cve.IDToParts(id)
+					cveSet.Add(cve)
+				}
+				count = len(cveSet)
+
+			} else {
+				count, err = m.imageCVEs.Count(ctx, fixableQuery)
+				if err != nil {
+					return nil, errors.Wrapf(err, "could not fetch cve component edge for cve %q for request %q", CVE, res.GetId())
```

Comment:
You should be able to use https://github.com/stackrox/stackrox/blob/042b99029c056239f5b244226654e81d26b77a6e/central/views/imagecve/view_impl.go#L32 to get the count for the old and the new.  Assuming that works, you wouldn't need this logic at all, just make a call to that function.